### PR TITLE
doc/rados: edit memory-profiling.rst

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/troubleshooting/memory-profiling.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 1]
```diff
@@ -2,16 +2,21 @@
  Memory Profiling
 ==================
 
-Ceph MON, OSD and MDS can generate heap profiles using
-``tcmalloc``. To generate heap profiles, ensure you have
-``google-perftools`` installed::
+Ceph MON, OSD and MDS can generate heap profiles using ``tcmalloc``. Make sure
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:5]: Ceph Monitor, OSD, and MDS daemons can report ``TCMalloc`` heap profiles.  Install ``google-perftools`` on the daemon systems if you wish to generate these.  Your OS distribution may package this under a different name, like ``gperftools``, and use a different package manager.

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 2]
```diff
@@ -60,112 +65,136 @@ in the interval. For instance::
      0.0   0.9%  97.7%      0.0  26.1% ReplicatedPG::do_op
      0.0   0.8%  98.5%      0.0   0.8% __gnu_cxx::new_allocator::allocate
 
-Refer to `Google Heap Profiler`_ for additional details.
+See `Google Heap Profiler`_ for additional details.
+
+After you have installed the heap profiler, start your cluster and begin using
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:70]: I think for most people the cluster will already be running.  I don't recall if the daemons need to restart after package installation.

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 3]
```diff
@@ -60,112 +65,136 @@ in the interval. For instance::
...
+that follow, do the following:
 
-Once you have the heap profiler installed, start your cluster and
-begin using the heap profiler. You may enable or disable the heap
-profiler at runtime, or ensure that it runs continuously. For the
-following commandline usage, replace ``{daemon-type}`` with ``mon``,
-``osd`` or ``mds``, and replace ``{daemon-id}`` with the OSD number or
-the MON or MDS id.
+#. replace ``{daemon-type}`` with ``mon``, ``osd`` or ``mds`` 
+#. replace ``{daemon-id}`` with the OSD number or the MON id or the MDS id
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:76]: ID

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 4]
```diff
@@ -60,112 +65,136 @@ in the interval. For instance::
...
-For example:: 
+.. prompt:: bash
 
-	ceph tell osd.0 heap stats
+   ceph tell osd.0 heap stats
 
-.. note:: Printing stats does not require the profiler to be running and does
-   not dump the heap allocation information to a file.
+.. note:: The printing of stats with this command does not require the profiler
+   to be running and does not dump the heap-allocation information to a file.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:113]: No hypen. s/printing/reporting/

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 5]
```diff
@@ -60,112 +65,136 @@ in the interval. For instance::
...
+----------------------------------------
 
 Running Massif heap profiler with Valgrind
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The Massif heap profiler tool can be used with Valgrind to
-measure how much heap memory is used and is good for
-troubleshooting for example Ceph RadosGW.
+The Massif heap profiler tool can be used with Valgrind to measure how much
+heap memory is used. This method is well-suited to troubleshooting RadosGW.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:176]: the Ceph Object Gateway?

File: doc/rados/troubleshooting/memory-profiling.rst [hunk 6]
```diff
@@ -60,112 +65,136 @@ in the interval. For instance::
...
 
-You can then run the ``ms_print`` command to get a graph and statistics
-from the collected data in the ``massif.out.<pid>`` file::
+.. prompt:: bash
 
-        ms_print massif.out.12345
+   ms_print massif.out.12345
 
-This output is great for inclusion in a bug report.
+The output of this command can often improve bug reports.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/memory-profiling.rst:200]: is helpful when submitting a bug report.

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/troubleshooting/memory-profiling.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/troubleshooting/memory-profiling.rst:5]: Accepted with slight modifications.

File: doc/rados/troubleshooting/memory-profiling.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/troubleshooting/memory-profiling.rst:76]: Accepted.

File: doc/rados/troubleshooting/memory-profiling.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/troubleshooting/memory-profiling.rst:113]: Accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53932 - Reef backport
https://github.com/ceph/ceph/pull/53933 - Quincy backport

**Pull request merged**